### PR TITLE
Consistent spacing between navbar and UI content

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

- Make VPNFlickable calculate the amount of padding to add to it's `contentHeight` to support an even amount of spacing between the bottom of each page's content and the navbar
- Remove a bunch of arbitrary padding from views

## Reference

[VPN-2875: Inconsistent content height in scrollable content](https://mozilla-hub.atlassian.net/browse/VPN-2875)

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

**Review by lesleyjanenorton — COMMENTED** [review]
Nice!

File: src/ui/screens/getHelp/giveFeedback/ViewGiveFeedbackThankYou.qml [hunk 1]
```diff
@@ -10,40 +10,42 @@ import Mozilla.VPN 1.0
...
-    id: root
-    _viewContentData: ColumnLayout {
+ColumnLayout {
+    id: layout
 
-        Layout.minimumHeight: root.height
+    spacing: 24
+
+    VPNPanel {
+        anchors.horizontalCenter: undefined
```
[the remark this answers is not in the document]
Comment (MattLichtenstein) [reply src/ui/screens/getHelp/giveFeedback/ViewGiveFeedbackThankYou.qml:19]: Always the iPad 😂

**Review by lesleyjanenorton — APPROVED** [review]

File: nebula/ui/components/VPNFlickable.qml [hunk 2]
```diff
@@ -64,7 +64,28 @@ Flickable {
         ensureVisAnimation.start();
     }
 
-    contentHeight: Math.max(window.safeContentHeight, flickContentHeight)
+    function recalculateContentHeight() {
+        //Checks if the flickable AND it's content interferes with the navbar area (bottom 146px for iOS, bottom 128px for other platforms)
+        //If it does, we pad the flickable's contentHeight with whatever bottom padding is needed so that there is always 48px (aka theme.navBarTopMargin)
+        //between the bottom of the flickable's content and the navbar
+        if (navbar.visible && mapToItem(window.contentItem, 0, 0).y + height >= window.height - VPNTheme.theme.navBarHeightWithMargins
+                && flickContentHeight + mapToItem(window.contentItem, 0, 0).y >= window.height - VPNTheme.theme.navBarHeightWithMargins) {
```
Comment (flozia) [inline nebula/ui/components/VPNFlickable.qml:72]: Nit: What do you think of assigning `mapToItem(window.contentItem, 0, 0).y` and `window.height - VPNTheme.theme.navBarHeightWithMargins` to a property with a descriptive name for better readability and reduced redundancy? They are used a couple of times in this file.

File: nebula/ui/components/VPNViewBase.qml [hunk 3]
```diff
@@ -28,32 +30,26 @@ Item {
...
         ColumnLayout {
             id: viewContent
             spacing: VPNTheme.theme.windowMargin
             anchors {
                 top: parent.top
-                topMargin: VPNTheme.theme.windowMargin
                 left: parent.left
                 right: parent.right
+                topMargin: VPNTheme.theme.windowMargin
+                bottomMargin: navbar.visible ? 0 : 40
```
Comment (flozia) [inline nebula/ui/components/VPNViewBase.qml:48]: Nit:
```suggestion
                bottomMargin: navbar.visible ? 0 : VPNTheme.theme.rowHeight
```

File: src/ui/screens/getHelp/contactUs/ViewContactUsThankYou.qml [hunk 4]
```diff
@@ -10,40 +10,43 @@ import Mozilla.VPN 1.0
...
-    property string _emailAddress: ""
+ColumnLayout {
     id: base
+    property string _emailAddress: ""
+
+    spacing: 24
+
+    VPNPanel {
+        anchors.horizontalCenter: undefined
+        Layout.topMargin: base.height * (window.fullscreenRequired() ? 0.3 :  0.2)
```
Comment (flozia) [inline src/ui/screens/getHelp/contactUs/ViewContactUsThankYou.qml:21]: Formatting nit:
```suggestion
        Layout.topMargin: base.height * (window.fullscreenRequired() ? 0.3 : 0.2)
```

File: src/ui/screens/getHelp/contactUs/ViewContactUsThankYou.qml [hunk 5]
```diff
@@ -10,40 +10,43 @@ import Mozilla.VPN 1.0
 import components 0.1
 import components.forms 0.1
 
-VPNViewBase {
-    property string _emailAddress: ""
+ColumnLayout {
     id: base
+    property string _emailAddress: ""
+
+    spacing: 24
```
Comment (flozia) [inline src/ui/screens/getHelp/contactUs/ViewContactUsThankYou.qml:17]: Nit:
```suggestion
    spacing: VPNTheme.theme.vSpacing
```

File: src/ui/screens/getHelp/giveFeedback/ViewGiveFeedbackThankYou.qml [hunk 6]
```diff
@@ -10,40 +10,42 @@ import Mozilla.VPN 1.0
 import components 0.1
 import components.forms 0.1
 
-VPNViewBase {
-    id: root
-    _viewContentData: ColumnLayout {
+ColumnLayout {
+    id: layout
+
+    spacing: 24
```
Comment (flozia) [inline src/ui/screens/getHelp/giveFeedback/ViewGiveFeedbackThankYou.qml:16]: Nit:
```suggestion
    spacing: VPNTheme.theme.vSpacing
```

File: src/ui/sharedViews/ViewErrorFullScreen.qml [hunk 7]
```diff
@@ -45,16 +44,11 @@ VPNFlickable {
...
-        anchors.bottom: parent.bottom
-        anchors.horizontalCenter: parent.horizontalCenter
-        spacing: 32
-
-        Component.onCompleted: {
-            height = Math.max(window.height, childrenRect.height)
-        }
+        anchors.fill: parent
+        anchors.leftMargin: 16
+        anchors.rightMargin: 16
```
Comment (flozia) [inline src/ui/sharedViews/ViewErrorFullScreen.qml:49]: Nit: `VPNTheme.theme.windowMargin`

**Review by flozia — APPROVED** [review]
Awesome! I only commented on nits that are are about formatting and theme variables. I tried to check as many screens as possible, but I’m sure I missed quite a few. Did not see any issues.

File: nebula/ui/components/VPNFlickable.qml (hunk 2 above)
Comment (MattLichtenstein) [reply nebula/ui/components/VPNFlickable.qml]: Good idea!

**Pull request merged**